packages/astro/src/content/vite-plugin-content-imports.ts

```diff
@@ -94,7 +94,7 @@ export function astroContentImportPlugin({
 					const code = escapeViteEnvReferences(`
 export const id = ${JSON.stringify(id)};
 export const collection = ${JSON.stringify(collection)};
-export const data = ${devalue.uneval(data) /* TODO: reuse astro props serializer */};
```

Comment:
Why _not_ use Astro's props serializer: it's overly forgiving for anything that isn't valid HTML. Example: it parses DayJS objects successfully, but silently strips all functions without an error. devalue is much more strict in what is and isn't supported, which should mean better error messages!